### PR TITLE
utfcpp: add version 4.0.5 and remove older versions

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **utfcpp/***

#### Motivation
Try to close https://github.com/conan-io/conan-center-index/issues/24760.

#### Details
https://github.com/nemtrif/utfcpp/compare/v4.0.4...v4.0.5

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — APPROVED** [review]
Thanks! I wonder if the installation mentioned in https://github.com/nemtrif/utfcpp?tab=readme-ov-file#installation could name Conan as a way to install, given that they are thinking about removing their upstream CMakeLists

File: recipes/utfcpp/all/conandata.yml [hunk 1]
```diff
@@ -11,27 +14,6 @@ sources:
   "3.2.5":
     url: "https://github.com/nemtrif/utfcpp/archive/v3.2.5.tar.gz"
     sha256: "14fd1b3c466814cb4c40771b7f207b61d2c7a0aa6a5e620ca05c00df27f25afd"
-  "3.2.4":
-    url: "https://github.com/nemtrif/utfcpp/archive/refs/tags/v3.2.4.tar.gz"
-    sha256: "fde21a4c519eed25f095a1cd8490167409cc70d7b5e9c38756142e588ccb7c7e"
-  "3.2.3":
-    url: "https://github.com/nemtrif/utfcpp/archive/v3.2.3.tar.gz"
-    sha256: "3ba9b0dbbff08767bdffe8f03b10e596ca351228862722e4c9d4d126d2865250"
```
Comment (ErniGH) [inline recipes/utfcpp/all/conandata.yml:19]: The 3.2.3 is being used by `lief`, `xlnt` and `antlr4-cppruntime`. We need to maintain it 😄

Comment (toge) [reply recipes/utfcpp/all/conandata.yml]: @ErniGH 
Thanks for your information!
I reverted it.

**Review by ErniGH — APPROVED** [review]
Thanks!

**Pull request merged**